ibm/ibm_key_protect_test.go

```diff
@@ -0,0 +1,104 @@
+package ibm
+
+import (
+	"os"
+	"testing"
+
+	"github.com/libopenstorage/secrets"
+	"github.com/libopenstorage/secrets/test"
+	"github.com/pborman/uuid"
+	"github.com/portworx/kvdb"
+	"github.com/portworx/kvdb/mem"
+)
+
+const (
+	testSecretId = "openstorage_secret"
+)
+
+func TestAll(t *testing.T) {
+	// Set the relevant environmnet fields for aws.
+	secretConfig := make(map[string]interface{})
+	// Fill in the appropriate keys and values
+	secretConfig[IbmServiceApiKey] = os.Getenv(IbmServiceApiKey)
+	secretConfig[IbmInstanceIdKey] = os.Getenv(IbmInstanceIdKey)
+	secretConfig[IbmCustomerRootKey] = os.Getenv(IbmCustomerRootKey)
+	secretConfig[IbmBaseUrlKey] = os.Getenv(IbmBaseUrlKey)
+	secretConfig[IbmTokenUrlKey] = os.Getenv(IbmTokenUrlKey)
+
+	// With kvdbPersistenceStore
+	kv, err := kvdb.New(mem.Name, "openstorage/", nil, nil, nil)
+	if err != nil {
+		t.Fatalf("Unable to create a AWS Secret instance: %v", err)
+		return
+	}
+	secretConfig[IbmKvdbKey] = kv
+	kp, err := New(secretConfig)
+	if err != nil {
+		t.Fatalf("Unable to create a IBM Key Protect Secret instance: %v", err)
+		return
+	}
+	ik := &ibmSecretTest{kp, ""}
+	test.Run(ik, t)
+}
```

Comment:
can you add test cases for New()?
you can look at vault_test.go or dcos_test.go

Reply:
done.